src/components/error-technical-difficulty/error-technical-difficulty.ts

```diff
@@ -0,0 +1,101 @@
+import moment from 'moment';
+import Vue, { PluginObject } from 'vue';
+import Component from 'vue-class-component';
+import { Prop } from 'vue-property-decorator';
+
+import { ModulVue } from '../../utils/vue/vue';
+import AccordionPlugin from '../accordion/accordion';
+import { ERROR_TECHNICAL_DIFFICULTY_NAME } from '../component-names';
+import ErrorTemplatePlugin, { Link, MErrorTemplateSkin } from '../error-template/error-template';
+import I18nPlugin from '../i18n/i18n';
+import LinkPlugin from '../link/link';
+import MessagePlugin from '../message/message';
+import PanelPlugin from '../panel/panel';
+import WithRender from './error-technical-difficulty.html?style=./error-technical-difficulty.scss';
+
+@WithRender
+@Component
+export class MErrorTechnicalDifficulty extends ModulVue {
+
+    @Prop({
+        default: () => (Vue.prototype as any).$i18n.translate('m-error-technical-difficulty:title')
+    })
+    public title: string;
+
+    @Prop({
+        default: () => [
+            new Link((Vue.prototype as any).$i18n.translate('m-error-technical-difficulty:help'), 'https://www.ene.ulaval.ca/contactez-nous?systeme_en_cours=81', true),
+            new Link((Vue.prototype as any).$i18n.translate('m-error-technical-difficulty:home-label'), '\\')]
+    })
+    public links: Link[];
+
+    @Prop({
+        default: () => [
+            (Vue.prototype as any).$i18n.translate('m-error-technical-difficulty:hint.primary'),
+            (Vue.prototype as any).$i18n.translate('m-error-technical-difficulty:hint.secondary')]
+    })
+    public hints: string[];
+
+    @Prop({
+        default: () => moment()
+    })
+    public errorDate: moment.Moment;
+
+    /**
+     * Reference number must be generated by the parent component
+     */
+    @Prop()
+    public errorReferenceNumber?: string;
+
+    @Prop({
+        default: false
+    })
+    public showStackTrace: boolean;
+
+    /**
+     * Javascript Error containing the stack trace to be displayed
+     */
+    @Prop()
+    public error?: Error;
+
+    readonly skin: string = MErrorTemplateSkin.Error;
+
+    readonly iconName: string = 'm-svg__error';
+
+    get userAgent(): string {
+        return window.navigator.userAgent;
+    }
+
+    /**
+     * Using the value of the props errorDate, generates an array with two values, the date in YYYY-MM-DD format and the time in HH:mm:ss format.
+     */
+    get dateInfo(): string[] {
+        let result: string[] = [];
+        if (this.errorDate) {
+            result = [this.errorDate.format('YYYY-MM-DD'), this.errorDate.format('HH:mm:ss')];
+        }
+        return result;
+    }
+
+    /**
+     * Defines if the stack trace is to be displayed based on the showStrackTrace prop and the presence of the attribute stack in the error.
+     */
+    get propStacktrace(): boolean {
+        return this.showStackTrace && !!this.error && !!this.error.stack;
```

Comment:
Ici, je mettrais juste `return this.showStackTrace && !!this.error;`. S'il n'y a pas de stack, j'aime mieux voir une boîte vide que de ne pas avoir de boîte et de me poser la question à savoir si la prop `showStackTrace` est bien à `true`.

Reply:
ok, mais normalement, pour brio en prod il n'y aura jamais de stack d'affichée à l'utilisateur, donc je trouvais ça poche d'afficher toujours une boite vide..

Reply:
Si en prod tu n'as jamais de stack, `this.showStackTrace` va être à `false` et tu n'auras pas la boîte vide dans ce cas

Reply:
a oui, pas trop allumé aujourd'hui!